faust/assignor/client_assignment.py

```diff
@@ -157,3 +159,11 @@ class ClientMetadata(Record,
     def __post_init__(self) -> None:
         if self.topic_groups is None:
             self.topic_groups = {}
+
+    def dumps(self) -> bytes:
+        return get_codec("json")._dumps(self.to_representation())
+
+    @classmethod
+    def loads(cls, s: bytes) -> Any:
+        data = get_codec("json")._loads(s)
+        return cls.from_data(data)
```

Comment:
Similar to below -- I wouldn't have expected changes to be needed to this file. Was this your way of testing the sometimes-await-sometimes-don't code?